pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1565,7 +1571,9 @@ public void deleteLedgerComplete(Object ctx) {
                 }).whenComplete((value, ex) -> {
                     if (ex != null) {
                         log.error("[{}] Error deleting topic", topic, ex);
-                        unfenceTopicToResume();
+                        if (!alreadyUnFenced.get()) {
+                            unfenceTopicToResume();
```

Comment:
I am not sure if this is a good practice to have a local var to control the unfenceTopicToResume() member func. Wouldn't it be cleaner if we can make this member func idempotent?(return early if the topic "state" is already unfenced.)

This brings back our earlier discussion that we probably need to refactor the topic state and its state handler logic.

Reply:
@heesung-sn 

Agree with you, we can merge the PR and then push a separated PR to improve it

Reply:
Merging the PR